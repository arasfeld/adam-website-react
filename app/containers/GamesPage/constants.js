/*
 * GamesPage Constants
 * 
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 */

export const LOAD_GAMES = 'app/Games/LOAD_GAMES';
export const LOAD_GAMES_SUCCESS = 'app/Games/LOAD_GAMES_SUCCESS';
export const LOAD_GAMES_ERROR = 'app/Games/LOAD_GAMES_ERROR';
