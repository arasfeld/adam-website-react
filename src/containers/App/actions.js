/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  CLOSE_SIDENAV,
  OPEN_SIDENAV,
  TOGGLE_SIDENAV,
} from './constants';

/**
 * Closes the sidenav on mobile, dispatched after...
 *
 * @return {object}  An action object with a type of CLOSE_SIDENAV
 */
export function closeSideNav() {
  return {
    type: CLOSE_SIDENAV,
  };
}

/**
 * Opens the sidenav on mobile, dispatched after clicking the hamburger icon in the header
 *
 * @return {object}  An action object with a type of OPEN_SIDENAV
 */
export function openSideNav() {
  return {
    type: OPEN_SIDENAV,
  };
}

/**
 * Toggles the sidenav on mobile, dispatched after clicking the hamburger icon in the header
 *
 * @return {object}  An action object with a type of TOGGLE_SIDENAV
 */
export function toggleSideNav() {
  return {
    type: TOGGLE_SIDENAV,
  };
}
