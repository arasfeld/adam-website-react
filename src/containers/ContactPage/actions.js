/*
 * Contact Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  CHANGE_EMAIL,
  CHANGE_NAME,
  CHANGE_TEXT,
  SEND_MESSAGE,
  SEND_MESSAGE_SUCCESS,
  SEND_MESSAGE_ERROR,
} from './constants';

/**
 * Changes the message input field of the form
 *
 * @param {object} email The new text of the message input field
 *
 * @return {object}    An action object with a type of CHANGE_MESSAGE
 */
export function changeEmail(email) {
  return {
    type: CHANGE_EMAIL,
    email,
  };
}

/**
 * Changes the message input field of the form
 *
 * @param {object} name The new text of the message input field
 *
 * @return {object}    An action object with a type of CHANGE_MESSAGE
 */
export function changeName(name) {
  return {
    type: CHANGE_NAME,
    name,
  };
}

/**
 * Changes the message input field of the form
 *
 * @param {object} text The new text of the message input field
 *
 * @return {object}    An action object with a type of CHANGE_MESSAGE
 */
export function changeText(text) {
  return {
    type: CHANGE_TEXT,
    text,
  };
}

/**
 * Send the message, this action starts the request saga
 *
 * @return {object} An action object with a type of SEND_MESSAGE
 */
export function sendMessage() {
  return {
    type: SEND_MESSAGE,
  };
}

/**
 * Dispatched when the message is sent by the request saga
 *
 * @return {object}      An action object with a type of SEND_MESSAGE_SUCCESS
 */
export function messageSent() {
  return {
    type: SEND_MESSAGE_SUCCESS,
  };
}

/**
 * Dispatched when sending the message fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of SEND_MESSAGE_ERROR passing the error
 */
export function messageSendingError(error) {
  return {
    type: SEND_MESSAGE_ERROR,
    error,
  };
}
