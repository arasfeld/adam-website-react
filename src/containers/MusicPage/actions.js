/*
 * Music Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_ALBUMS,
  LOAD_ALBUMS_SUCCESS,
  LOAD_ALBUMS_ERROR,
  LOAD_ARTISTS,
  LOAD_ARTISTS_SUCCESS,
  LOAD_ARTISTS_ERROR,
  LOAD_LAST_TRACK,
  LOAD_LAST_TRACK_SUCCESS,
  LOAD_LAST_TRACK_ERROR,
} from './constants';

/**
 * Load the albums, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_ALBUMS
 */
export function loadAlbums() {
  return {
    type: LOAD_ALBUMS,
  };
}

/**
 * Dispatched when the albums are loaded by the request saga
 *
 * @param  {array} albums The album data
 *
 * @return {object}      An action object with a type of LOAD_ALBUMS_SUCCESS passing the albums
 */
export function albumsLoaded(albums) {
  return {
    type: LOAD_ALBUMS_SUCCESS,
    albums,
  };
}

/**
 * Dispatched when loading the albums fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_ALBUMS_ERROR passing the error
 */
export function albumLoadingError(error) {
  return {
    type: LOAD_ALBUMS_ERROR,
    error,
  };
}

/**
 * Load the artists, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_ARTISTS
 */
export function loadArtists() {
  return {
    type: LOAD_ARTISTS,
  };
}

/**
 * Dispatched when the artists are loaded by the request saga
 *
 * @param  {array} artists The album data
 *
 * @return {object}      An action object with a type of LOAD_ARTISTS_SUCCESS passing the artists
 */
export function artistsLoaded(artists) {
  return {
    type: LOAD_ARTISTS_SUCCESS,
    artists,
  };
}

/**
 * Dispatched when loading the artists fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_ARTISTS_ERROR passing the error
 */
export function artistLoadingError(error) {
  return {
    type: LOAD_ARTISTS_ERROR,
    error,
  };
}

/**
 * Load the last played track, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_LAST_TRACK
 */
export function loadLastTrack() {
  return {
    type: LOAD_LAST_TRACK,
  };
}

/**
 * Dispatched when the track is loaded by the request saga
 *
 * @param  {array} track The track data
 *
 * @return {object}      An action object with a type of LOAD_LAST_TRACK_SUCCESS passing the track
 */
export function lastTrackLoaded(track) {
  return {
    type: LOAD_LAST_TRACK_SUCCESS,
    track,
  };
}

/**
 * Dispatched when loading the track fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_LAST_TRACK_ERROR passing the error
 */
export function lastTrackLoadingError(error) {
  return {
    type: LOAD_LAST_TRACK_ERROR,
    error,
  };
}
