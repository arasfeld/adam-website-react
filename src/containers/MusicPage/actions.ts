import {
  LOAD_MUSIC,
  LOAD_MUSIC_SUCCESS,
  LOAD_MUSIC_ERROR,
  Album,
  Artist,
  Track,
  LoadMusic,
  LoadMusicSuccess,
  LoadMusicError,
} from './types';

/**
 * Load the music data, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_MUSIC
 */
export function loadMusic(): LoadMusic {
  return {
    type: LOAD_MUSIC,
  };
}

/**
 * Dispatched when the music data is loaded by the request saga
 *
 * @param  {array} albums The album data
 * @param  {array} artists The artist data
 * @param  {array} tracks The track data
 *
 * @return {object}      An action object with a type of LOAD_MUSIC_SUCCESS passing the music data
 */
export function musicLoaded(albums: Album[], artists: Artist[], tracks: Track[]): LoadMusicSuccess {
  return {
    type: LOAD_MUSIC_SUCCESS,
    albums,
    artists,
    tracks,
  };
}

/**
 * Dispatched when loading the music data fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_MUSIC_ERROR passing the error
 */
export function musicLoadingError(error: any): LoadMusicError {
  return {
    type: LOAD_MUSIC_ERROR,
    error,
  };
}
