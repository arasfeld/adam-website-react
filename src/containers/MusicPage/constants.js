/*
 * MusicConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const LOAD_ALBUMS = 'app/Music/LOAD_ALBUMS';
export const LOAD_ALBUMS_SUCCESS = 'app/Music/LOAD_ALBUMS_SUCCESS';
export const LOAD_ALBUMS_ERROR = 'app/Music/LOAD_ALBUMS_ERROR';

export const LOAD_ARTISTS = 'app/Music/LOAD_ARTISTS';
export const LOAD_ARTISTS_SUCCESS = 'app/Music/LOAD_ARTISTS_SUCCESS';
export const LOAD_ARTISTS_ERROR = 'app/Music/LOAD_ARTISTS_ERROR';
