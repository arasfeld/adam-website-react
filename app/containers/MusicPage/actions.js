/*
 * Music Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import { LOAD_MUSIC, LOAD_MUSIC_SUCCESS, LOAD_MUSIC_ERROR } from './constants';

/**
 * Load the music data, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_MUSIC
 */
export function loadMusic() {
  return {
    type: LOAD_MUSIC,
  };
}

/**
 * Dispatched when the music data is loaded by the request saga
 *
 * @param  {array} albums The album data
 * @param  {array} artists The artist data
 * @param  {array} tracks The track data
 *
 * @return {object}      An action object with a type of LOAD_MUSIC_SUCCESS passing the music data
 */
export function musicLoaded(albums, artists, tracks) {
  return {
    type: LOAD_MUSIC_SUCCESS,
    albums,
    artists,
    tracks,
  };
}

/**
 * Dispatched when loading the music data fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_MUSIC_ERROR passing the error
 */
export function musicLoadingError(error) {
  return {
    type: LOAD_MUSIC_ERROR,
    error,
  };
}
