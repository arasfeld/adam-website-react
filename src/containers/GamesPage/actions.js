/*
 * GamesPage Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 */

import { LOAD_GAMES, LOAD_GAMES_SUCCESS, LOAD_GAMES_ERROR } from './constants';

/**
 * Load the game data, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_GAMES
 */
export function loadGames() {
  return {
    type: LOAD_GAMES,
  };
}

/**
 * Dispatched when the game data is loaded by the request saga
 *
 * @param  {array} ownedGames The owned game data
 * @param  {array} recentlyPlayedGames The recently played game data
 *
 * @return {object} An action object with a type of LOAD_GAMES_SUCCESS passing the game data
 */
export function gamesLoaded(ownedGames, recentlyPlayedGames) {
  return {
    type: LOAD_GAMES_SUCCESS,
    ownedGames,
    recentlyPlayedGames,
  };
}

/**
 * Dispatched when loading the games fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of LOAD_GAMES_ERROR passing the error
 */
export function gamesLoadingError(error) {
  return {
    type: LOAD_GAMES_ERROR,
    error,
  };
}
