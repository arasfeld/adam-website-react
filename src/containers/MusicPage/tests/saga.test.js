import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_MUSIC } from '../constants';
import { musicLoaded, musicLoadingError } from '../actions';

import lastFmData, { getMusic } from '../saga';

describe('getMusic Saga', () => {
  let getMusicGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getMusicGenerator = getMusic();

    const callDescriptor = getMusicGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the musicLoaded action if it requests the data successfully', () => {
    const response = {
      albums: [],
      artists: [],
      tracks: [],
    };
    const putDescriptor = getMusicGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(musicLoaded(response)));
  });

  it('should dispatch the musicLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getMusicGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(musicLoadingError(response)));
  });
});

describe('lastFmData Saga', () => {
  const lastFmDataSaga = lastFmData();

  it('should start task to watch for LOAD_MUSIC actions', () => {
    const takeLatestLoadMusicDescriptor = lastFmDataSaga.next().value;
    expect(takeLatestLoadMusicDescriptor).toEqual(takeLatest(LOAD_MUSIC, getMusic));
  });
});
