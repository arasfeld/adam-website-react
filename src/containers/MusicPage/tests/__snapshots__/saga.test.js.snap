// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMusic Saga should dispatch the musicLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://ws.audioscrobbler.com/2.0/?method=user.gettopalbums&user=arazzy&api_key=API_KEY&limit=21&format=json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getMusic Saga should dispatch the musicLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://ws.audioscrobbler.com/2.0/?method=user.gettopartists&user=arazzy&api_key=API_KEY&limit=21&format=json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getMusic Saga should dispatch the musicLoaded action if it requests the data successfully 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://ws.audioscrobbler.com/2.0/?method=user.getrecenttracks&user=arazzy&api_key=API_KEY&limit=10&format=json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getMusic Saga should dispatch the musicLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://ws.audioscrobbler.com/2.0/?method=user.gettopalbums&user=arazzy&api_key=API_KEY&limit=21&format=json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getMusic Saga should dispatch the musicLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://ws.audioscrobbler.com/2.0/?method=user.gettopartists&user=arazzy&api_key=API_KEY&limit=21&format=json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getMusic Saga should dispatch the musicLoadingError action if the response errors 3`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://ws.audioscrobbler.com/2.0/?method=user.getrecenttracks&user=arazzy&api_key=API_KEY&limit=10&format=json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
